### PR TITLE
chore(internal): start crashtracker when enabled

### PR DESCRIPTION
**Pull request by brettlangdon**
WIP: initial implementation, needs some manual and automated testing changes.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/internal/crashtracker/test_crashtracker.py [hunk 1]
```diff
@@ -274,3 +275,141 @@ def test_crashtracker_raise_sigbus():
...
+    # Setup the listening socket before we open ddtrace
+    port, sock = utils.crashtracker_receiver_bind()
+    assert sock
+    os.putenv("DD_TRACE_AGENT_URL", "http://localhost:%d" % port)
+
+    # Call the program
+    root_dir = os.environ["TEST_FILE_ROOT"]
+    stdout, stderr, exitcode, _ = call_program(
+        "ddtrace-run", sys.executable, os.path.join(root_dir, "simple_crashing_preload.py")
+    )
```
Comment (brettlangdon) [inline tests/internal/crashtracker/test_crashtracker.py:302]:
```python
def test_crashtracker_preload_default(ddtrace_run_python_code_in_subprocess):
    code = """
import ctypes


ctypes.string_at(0)
exit(-1)
    """

    stdout, stderr, exitcode, _ = ddtrace_run_python_code_in_subprocess(code, env=env)
```

`run_python_code_in_subprocess` and `ddtrace_run_python_code_in_subprocess` might help clean this up a little bit, at least not needing to know current root dir, or creating dummy/test files.


what you have is totally fine, just sharing an alternative possible approach. this is a non-blocking comment/no change is needed.

Comment (sanchda) [reply tests/internal/crashtracker/test_crashtracker.py:302]: Oh cool, TIL.  Will add a patch for that

Comment (sanchda) [reply tests/internal/crashtracker/test_crashtracker.py]: fixed in [219a3a1](https://github.com/DataDog/dd-trace-py/pull/9865/commits/219a3a13927fe41acca3ff2797a467c7aeb415b2)

**Review by sanchda — APPROVED** [review]

**Review by brettlangdon — COMMENTED** [review]
choosing to not add a release note since this is not a public facing feature. we do have some config toggles for people (like disabling the feature), but for now this is an internal only used featured.

**Review by emmettbutler — APPROVED** [review]

**Review by taegyunkim — APPROVED** [review]
LGTM, Approval for ddtrace/internal/datadog/profiling/crashtracker/__init__.py

**Pull request merged**